### PR TITLE
Create Task threads before scheduling tasks

### PR DESCRIPTION
**Pull request by vmamidi**
Tasks are scheduled on NET threads as there are no task threads available when the tasks are scheduled.

### CODE REVIEW

**Comment by vmamidi** [comment]
[approve ci autest]

**Comment by vmamidi** [comment]
[approve ci autest]

**Comment by randall** [comment]
These are legit autest failures. At least one of these tests, has a global plugin that attempts to configure logging for itself, but crashes because logging hasn't been initialized yet.

**Comment by randall** [comment]
[approve ci fedora]

**Comment by randall** [comment]
[approve ci fedora]
[approve ci autest]

**Comment by randall** [comment]
[approve ci fedora]

**Comment by randall** [comment]
[approve ci autest]

**Comment by randall** [comment]
[approve ci]

**Comment by bryancall** [comment]
[approve ci]

File: src/traffic_server/traffic_server.cc [hunk 1]
```diff
@@ -2101,7 +2104,7 @@ main(int /* argc ATS_UNUSED */, const char **argv)
 
     // Init plugins as soon as logging is ready.
     (void)plugin_init(); // plugin.config
-
+    task_threads_started_callback();
```
Comment (SolidWallOfCode) [inline src/traffic_server/traffic_server.cc:2107]: Why move this?

**Review by duke8253 — COMMENTED** [review]
Maybe we should change the name of that function to be just `task_threads_started()` since it's not a callback anymore? Just nitpicking.

**Comment by randall** [comment]
[approve ci autest]

**Comment by ywkaras** [comment]
[approve ci autest]

**Comment by duke8253** [comment]
[approve ci autest]

**Comment by duke8253** [comment]
Noticed something interesting. The test is failing because after the change, it no longer runs on a `ET_TASK` thread, instead it runs on the main thread for some reason.

**Comment by bneradt** [comment]
Hi @vmamidi . Just making sure you saw @duke8253's message above. I've pulled in his fix internally at Yahoo and CI has been happy with it. It should fix things here.

**Comment by vmamidi** [comment]
[approve ci autest]

**Comment by duke8253** [comment]
This is fixed by PR #8981

**Comment by zwoop** [comment]
Remember to remove Milestone's and Project's when closing a PR without merging it.

**Pull request closed without merging**